modules/drivers/axp192/axp192.js

```diff
@@ -17,34 +17,141 @@
  *
  */
-import I2C from 'pins/i2c';
+import SMBus from "pins/smbus";
 
-export default class AXP192 extends I2C {
-	constructor(it) {
-		super(it);
-		this.initialize();
+class DCDC {
+  #parent;
+  #register;
+  constructor({ register, parent }) {
+    this.#parent = parent;
+    this.#register = register;
+  }
+  set voltage(v) {
+    const vdata = v < 700 ? 0 : (v - 700) / 25;
+    this.#parent.writeByte(
+      this.#register,
+      (this.#parent.readByte(this.#register) & 0x80) | (vdata & 0x7f)
+    );
+  }
+  get voltage() {
+    return (this.#parent.readByte(this.#register) & 0x7f) * 25 + 700;
+  }
+}
+
+class LDO {
+  #parent;
+  #register;
+  #offsetV;
+  #offsetEn;
+  constructor({ register, parent, offsetV, offsetEn }) {
+    this.#parent = parent;
+    this.#register = register;
+    this.#offsetV = offsetV;
+    this.#offsetEn = offsetEn;
+  }
+
+  set voltage(v) {
+    const vdata = v > 3300 ? 15 : v / 100 - 18;
+    const mask = ~(0xff << this.#offsetV);
+    this.#parent.writeByte(
+      this.#register,
+      (this.#parent.readByte(this.#register) & mask) | (vdata << this.#offsetV)
+    );
+  }
+
+  get voltage() {
+    return (
+      ((this.#parent.readByte(this.#register) >> this.#offsetV) + 18) * 100
+    );
+  }
+
+  set enable(enable) {
+    const mask = 0x01 << this.#offsetEn;
+    if (enable) {
+      this.#parent.writeByte(0x12, this.#parent.readByte(0x12) | mask);
+    } else {
+      this.#parent.writeByte(0x12, this.#parent.readByte(0x12) & ~mask);
+    }
+  }
+
+  get enable() {
+    return Boolean((this.#parent.readByte(0x12) >> this.#offsetEn) & 1);
+  }
+}
+
+class GPIO {
+	#register
+	#parent
+	#mask
+	constructor({ register, parent, offset }) {
+    this.#parent = parent;
+		this.#register = register;
+		this.#mask = 0x01 << offset;
 	}
 
-	/**
-	 * initialize axp192
-	 */
-	initialize() {
-		this.write(0x10, 0xff); // OLED VPP Enable
-		this.write(0x28, 0xff); // Enable LDO2&LDO3, LED&TFT 3.3V
-		this.write(0x82, 0xff); // Enable all the ADCs
-		this.write(0x33, 0xc0); // Enable Charging, 100mA, 4.2V End at 0.9
-		this.write(0xB8, 0x80); // Enable Colume Counter
-		this.write(0x12, 0x4d); // Enable DC-DC1, OLED VDD, 5B V EXT
-		this.write(0x36, 0x5c); // PEK
-		this.write(0x90, 0x02); // gpio0
+	get enable() {
+		return Boolean(this.#parent.readByte(this.#register) & this.#mask);
 	}
 
-	/**
-	 * sets the screen brightness
-	 * @param {*} brightness brightness between 7-15
-	 */
-	setBrightness(brightness) {
-		const b = (brightness & 0x0f) << 4;
-		this.write(0x28, b);
+	set enable(enable) {
+    let data = this.#parent.readByte(this.#register);
+    if (enable) {
+      data |= this.#mask;
+    } else {
+      data &= ~this.#mask;
+    }
+    this.#parent.writeByte(this.#register, data);
 	}
 }
+
+export default class AXP192 extends SMBus {
+  constructor(it) {
+    if (it.address == null) {
+      it.address = 0x34;
+    }
+    super(it);
```

Comment:
It is preferred to not modify the input argument, as it may be frozen or the caller may not expect changes. Here's a concise way to do that.

```js
super({address: 0x34, ...it)}
```

Reply:
Pushed a fix according above.